config/routes.rb

```diff
@@ -0,0 +1,75 @@
+Rails.application.routes.draw do
+
+  root to: 'tasks#index'
+
+  get 'tasks/index', as: 'index'
+
+  get 'tasks/show/:id' => 'tasks#show', as: 'show'
+
+  post 'tasks/create' => 'tasks#create', as: 'create'
+
+  get 'tasks/new' => 'tasks#new', as: 'new'
+
+  get 'tasks/:id/edit' => 'tasks#edit', as: 'edit'
+
+  patch 'tasks/:id/update' => 'tasks#update', as: 'update'
+
+  patch 'tasks/:id/show' => 'tasks#button', as: 'button'
```

Comment:
@brianaeng this "button" route looks a little confusing along with the normal show route.